orca-run-multiple-pipelines/src/main/java/io/armory/plugin/smp/execution/MyExecutionLauncher.java

```diff
@@ -78,19 +76,21 @@ public MyExecutionLauncher(
         this.clock = clock;
         this.applicationEventPublisher = applicationEventPublisher;
         this.pipelineValidator = pipelineValidator;
-        this.registry = registry;
+        this.executionConfigurationProperties = executionConfigurationProperties;
     }
 
-    public PipelineExecution start(ExecutionType type, String configJson) throws Exception {
-        final PipelineExecution execution = parse(type, configJson);
-
+    @Override
+    public PipelineExecution start(ExecutionType type, Map<String, Object> config) {
+        PipelineExecution execution = parse(type, config);
         PipelineExecution existingExecution = checkForCorrelatedExecution(execution);
+
         if (existingExecution != null) {
-            if (configJson.contains("correlationId") && configJson.contains("parentExecution")
-                    && configJson.contains("\"type\":\"runMultiplePipelines\"")) {
+            if (config.containsKey("correlationId") && config.containsKey("parentExecution")
+                    && config.containsKey("\"type\":\"runMultiplePipelines\"")) {
```

Comment:
can we add a unit test to see if this actually detects a key like this? especially in a nested object?